pydust/src/pytypes.zig

```diff
@@ -284,3 +306,31 @@ fn Slots(comptime definition: type, comptime Instance: type) type {
         }
     };
 }
+
+fn BinaryOperator(
+    comptime definition: type,
+    comptime Instance: type,
+    comptime op: []const u8,
+    comptime op_slot: c_int,
+) type {
+    return struct {
+        const slot: c_int = op_slot;
+
+        fn call(pyself: *ffi.PyObject, pyother: *ffi.PyObject) callconv(.C) ?*ffi.PyObject {
+            const func = @field(definition, op);
+            const typeInfo = @typeInfo(@TypeOf(func));
+            const sig = funcs.parseSignature(op, typeInfo.Fn, &.{});
+
+            if (sig.selfParam == null) @compileError(op ++ " must take a self parameter");
+            if (sig.nargs != 1) @compileError(op ++ " must take exactly one parameter after self parameter");
+
+            const self: *Instance = @ptrCast(pyself);
+            const other = tramp.Trampoline(
+                sig.argsParam orelse unreachable,
+            ).unwrap(.{ .py = pyother }) catch return null;
+
+            const result = func(&self.state, other) catch return null;
```

Comment:
Do you not need to trampoline the self argument?

Reply:
We don't do it like that in any of the methods, e.g. __len__